msp/mspimpl.go

```diff
@@ -889,28 +889,42 @@ func (msp *bccspmsp) getCertificationChainIdentifierFromChain(chain []*x509.Cert
 // do have signatures in Low-S. If this is not the case, the certificate
 // is regenerated to have a Low-S signature.
 func (msp *bccspmsp) sanitizeCert(cert *x509.Certificate) (*x509.Certificate, error) {
+	var err error
 	if isECDSASignedCert(cert) {
+		isRootCACert := false
+		validityOpts := msp.getValidityOptsForCert(cert)
+		if cert.CheckSignatureFrom(cert) == nil && cert.IsCA {
```

Comment:
Here we anyway verify a signature even if it's not a CA certificate.

It makes more sense to switch the order of conditions of either side of the && so that if it's not a CA we won't bother verifying the signature.

Reply:
Pushed fix, thanks.